aten/src/ATen/native/mps/operations/UnaryKernel.mm

```diff
@@ -25,6 +25,14 @@ static void round_decimals_kernel(TensorIteratorBase& iter, int64_t decimals) {
   lib.exec_unary_kernel(iter, "round_decimals", Scalar(decimals), ScalarType::Long);
 }
 
+static void pow_tensor_scalar_kernel(TensorIteratorBase& iter, const Scalar& exp_scalar) {
+  if (exp_scalar.to<int>() == 2) {
```

Comment:
Ugh what if exp_scalar isn't an int? Will this round down?

Reply:
I need to add a check there (and I hope we have tests for it), because `torch.randint(-10, 10, (3, )).pow(2)` returns tensor of longs, but `torch.randint(-10, 10, (3, )).pow(2.0)` yields a tensor of floats